### PR TITLE
Improve Reweights

### PR DESCRIPTION
**Pull request by Jeff-Thompson12**
PR for continued improvements to #208. Addresses issue #220 as well.

### CODE REVIEW

**Comment by AARON-CLARK** [comment]
Is this one ready for review too @Jeff-Thompson12 ?

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK it addresses both items you noted in #220, but you did say it was not a comprehensive list. So I was looking for what other improvements are needed. Those could be handled in a separate PR though.

It makes sense to go ahead and review it so that those changes can be implemented before our meeting.

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
Hi @Jeff-Thompson12, I don't think this has to be completed by our meeting today, so I'll request changes. Couple of things I noticed:

(1) First, the whole re-weighting "module" should be moved to the Admin tools tab. So, it may be appropriate to just make a tabsetPanel, one tabPanel for the "credential manager" and another tabPanel for the "assessment criteria"?

(2) Are you getting double column sorter icon's on the table like me? It looks like they didn't exist when I took the screenshot in issue #220. See image below:
image

(3) It looks like if you submit a final decision for package, then change the weights and re-calc risk scores, the currently selected package decision doesn't get reverted/ undone until you toggle the selected package in the sidebar (to anything) and then re-select the previous package that formerly had a decision made. Can package decision be more reactive in this context?

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK this is ready for you to take another look.

**Comment by AARON-CLARK** [comment]
Hi @Jeff-Thompson12, there are some conflicts on this branch now.

**Review by AARON-CLARK — CHANGES_REQUESTED** [review]
@Jeff-Thompson12, looks great! However, I set a package's risk to 'medium' and then re-calculated risk scores and the slider went back down to 'low' on the sidebar. After thinking about it more this morning, I don't really care about maintaining that slider's previous value, so I would just forget about it unless you really want to fix it. Instead, I have a few small aesthetic requests:

I think it's slightly hard to see both of these tabs (pictured below) since they are spanning the whole screen. And the bootstrap theme we chose doesn't help since makes them both white (even when one is selected... usually the shading changes when selected). Do you think you could make the tabs the approximate size of the text string. Also add just a little margin between the title and the start of the tabs. Maybe just one or two `br()`s.

image

Second, the message/note at the bottom of the metric re-weighting needs some margin around it too. Maybe a couple more `br()`s and put some padding on the left-hand side if you think it looks good.

image

Is it possible to have the name of the package show up on the second line? I did this for package upload tab's pop-up and I think it looks really good!

image

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK the packages are not on the second line for me in the package upload. The `\n` characters seem to be getting ignored in `withProgress` but I will look into it more.

**Comment by AARON-CLARK** [comment]
> @AARON-CLARK the packages are not on the second line for me in the package upload. The `\n` characters seem to be getting ignored in `withProgress` but I will look into it more.

Ya, I think I noticed that too, so I just made the pop-up box smaller, constrained to a certain number of pixels. But it sounds like that's not working for you! lol shoot. Let me know what you find.

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK you can't manually add the break in the text itself because the progress message and details are spans. I just added a little JS to add a <br> HTML element after the progress message within the withProgress.

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK this PR is ready for you to take another look

File: R/reweightView.R [hunk 1]
```diff
@@ -101,12 +101,15 @@ reweightViewServer <- function(id, user) {
...
               fluidRow(
-                column(width = 12,
+                column(width = 1),
+                column(width = 10,
                        h5(em("Note: Changing the weights of the metrics will not update the
                risk of the packages on the database until 'Re-calculate' button is selected.
-               ")))
+               ")), align = "center"),
+                column(width = 1)
               )
```
[the remark this answers is not in the document]
Comment (AARON-CLARK) [reply R/reweightView.R:113]: line 113?

**Comment by AARON-CLARK** [comment]
Hi @Jeff-Thompson12, not trying to be knit-picky, but now my message is text wrapping, then displaying the package name. Is this happening for you? If you're good with it then I'm good with it.

**Comment by Jeff-Thompson12** [comment]
> Hi @Jeff-Thompson12, not trying to be knit-picky, but now my message is text wrapping, then displaying the package name. Is this happening for you? If you're good with it then I'm good with it.
> 
> image

The message for the progress notification was always wrapping for me.

File: R/reweightView.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (AARON-CLARK) [reply R/reweightView.R:113]: @Jeff-Thompson12, correct. I see you put some `br()`s before the note, but I don't think those were necessary (unless you like them there). I was only envisioning added margin after the note. Thanks!

**Comment by Jeff-Thompson12** [comment]
@AARON-CLARK I used jQuery to adjust the width of the `shiny-notification` and added padding beneath the notice.

**Review by AARON-CLARK — APPROVED** [review]
beautiful!

**Pull request merged**